client/driver/docker.go

```diff
@@ -153,8 +153,14 @@ func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) d
 		hostConfig.PortBindings = dockerPorts
 	}
 
+	// Merge Nomad-native with user-specified envvars
+	env := TaskEnvironmentVariables(ctx, task).List()
+	for k, v := range task.Env {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
```

Comment:
Perhaps pass `v` through  `strings.ToUpper` similar to how the Meta is processed?